### ISSUE TITLE
Have SFTP PSProvider

### ISSUE THREAD

**Issue by SteveL-MSFT** [PowerShell/PowerShell#9505]
# Summary of the new feature/enhancement

https://github.com/libfuse/sshfs enables mounting a SFTP endpoint as a local filesystem.  Seems like we should be able to use this concept with a PSProvider thus enabling something like:

```powershell
cd SFTP:
# no creds passed as using pubkey auth
SFTP:> Connect-SftpServer -HostName foo
cd foo
SFTP:/foo>
```

# Proposed technical implementation details (optional)

Since we want this to be cross platform, we shouldn't use `sshfs` linked above.  Instead, we should be able to just use `sftp.exe` directly with some check that it is available.  This will work on Windows with OpenSSH installed.

**Comment by strawgate** [comment]
This would be great if it also supported insecure ftp

**Comment by fMichaleczek** [comment]
@strawgate This enhancement is about FTP over SSH (SFTP) to improve remote administration. In my opinion, FTP and FTPS are out of scope and need a different library (like this one : [FluentFTP](https://github.com/robinrodricks/FluentFTP) )

**Comment by iSazonov** [comment]
FTP is very old, insecure protocol with very poor capabilities. I see use cases for compatibilites with old systems which can work only with FTP but I think this is the case when it should be separated from core.
For reference. We should work transparently with any transport (WSMan, SSH and etc). Also we discussed previously that we need to implement provider-to-provider communications (ex.: copy-item).

**Comment by SteveL-MSFT** [comment]
Regular FTP should be covered by https://github.com/PowerShell/PowerShell-RFC/pull/124

**Comment by chuanjiao10** [comment]
very good!!!

cd foo
[SFTP] foo:\>copy-item ; new-item; ...

**Comment by SteveL-MSFT** [comment]
With such a provider, you don't even have to `cd` into the psdrive, you should be able to do this:

```powershell
copy-item sftp:/foo/path/file ~/Downloads/
```

**Comment by ThomasNieto** [comment]
If this gets implemented #6586 would need to be fixed to enable cross-provider support for the example to work.

**Comment by chrgraefe** [comment]
@SteveL-MSFT: Are the any ideas to put some effort in this implementation?
I don't understand your proposed technical idea. Should this work as a wrapper around the ssh and sftp commands?
